sources/packages/web/src/components/aest/students/bypassedRestrictions/BypassRestrictionModal.vue

```diff
@@ -0,0 +1,280 @@
+    const availableStudentRestrictions = ref(
+      {} as AvailableStudentRestrictionsAPIOutDTO,
+    );
+    const {
+      showDialog,
+      showModal: showModalInternal,
+      resolvePromise,
+      loading,
+    } = useModalDialog<boolean>();
+    const bypassRestrictionForm = ref({} as VForm);
+    const { checkNullOrEmptyRule, checkNotesLengthRule } = useRules();
+    const note = ref("");
+    const cancel = () => {
+      restrictionBypassDetails.value =
+        {} as ApplicationRestrictionBypassAPIOutDTO;
+      bypassRestrictionForm.value.reset();
+      resolvePromise(false);
+    };
+    const bypassRestriction = async () => {
+      const validationResult = await bypassRestrictionForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      try {
+        loading.value = true;
+        await ApplicationRestrictionBypassService.shared.bypassRestriction({
+          applicationId: applicationId.value,
+          studentRestrictionId: formModel.studentRestrictionId,
+          bypassBehavior: formModel.bypassBehavior,
+          note: formModel.note,
+        });
+        snackBar.success("Restriction bypassed.");
+        resolvePromise(true);
+      } catch (error: unknown) {
+        if (error instanceof ApiProcessError) {
+          snackBar.warn(error.message);
+        } else {
+          snackBar.error(
+            "An unexpected error happen while bypassing the restriction.",
+          );
+        }
+      } finally {
+        loading.value = false;
+      }
+    };
+
+    const getRestrictionToBypassOption = (
+      restrictionCode: string,
+      createdDate: Date,
+      studentRestrictionId: number,
+    ) => {
+      const formattedDate = dateOnlyLongString(createdDate);
+      return {
+        title: `${restrictionCode} added on ${formattedDate}`,
+        value: studentRestrictionId,
+      };
+    };
+
+    const showModal = async (params: {
+      applicationId?: number;
+      applicationRestrictionBypassId?: number;
+    }) => {
+      if (params.applicationId) {
+        readOnly.value = false;
+        formModel.note = "";
+        applicationId.value = params.applicationId;
+        availableStudentRestrictions.value =
+          await ApplicationRestrictionBypassService.shared.getAvailableStudentRestrictions(
+            applicationId.value,
+          );
+        if (!availableStudentRestrictions.value.availableRestrictionsToBypass)
+          return;
+        restrictionsToBypass.value =
+          availableStudentRestrictions.value.availableRestrictionsToBypass.map(
+            (restriction) => {
+              return getRestrictionToBypassOption(
+                restriction.restrictionCode,
+                restriction.studentRestrictionCreatedAt,
+                restriction.studentRestrictionId,
+              );
+            },
+          );
+      }
+      if (params.applicationRestrictionBypassId) {
```

Comment:
Are the `params` optional properties mutually exclusive?
If yes an `else` can be added.